### PR TITLE
PIP-913-add-pe-trim-align-mark-filter

### PR DESCRIPTION
**Pull request by keenangraham**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -83,7 +83,11 @@ jobs:
             echo "pulling template!"
             docker pull quay.io/encode-dcc/dnase-seq-pipeline:template
             docker login -u=${QUAY_ROBOT_USER} -p=${QUAY_ROBOT_USER_TOKEN} quay.io
-            docker build --cache-from quay.io/encode-dcc/dnase-seq-pipeline:template --build-arg GIT_COMMIT_HASH=${CIRCLE_SHA1} --build-arg BRANCH=${CIRCLE_BRANCH} --build-arg BUILD_DNASE_DOCKER_IMAGE_TAG=${DNASE_DOCKER_IMAGE_TAG} -t $DNASE_DOCKER_IMAGE_TAG .
+            docker build --cache-from quay.io/encode-dcc/dnase-seq-pipeline:template \
```
Comment (ottojolanki) [inline .circleci/config.yml]: This sparks joy.

File: wdl/workflows/pe/align.wdl [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        input:
+            indexed_fasta=indexed_fasta,
+            sam=make_sam_from_sai_and_fastq_pair.sam,
+            resources=compute.runtimes[machine_size],
+    }
+
+    call unsorted_bam.sort_bam_with_samtools {
+        input:
+            bam=convert_sam_to_bam.unsorted_bam,
+            resources=compute.runtimes[machine_size],
```
Comment (ottojolanki) [inline wdl/workflows/pe/align.wdl]:
```suggestion
            resources=compute.runtimes[machine_size],
```

File: wdl/workflows/pe/trim.wdl [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+workflow trim {
+    input {
+        FastqPair raw_fastqs
+        File adapters
+        Int trim_length = 101
+        String machine_size
+    }
+
+    Machines compute = read_json("wdl/runtimes.json")
+
```
Comment (ottojolanki) [inline wdl/workflows/pe/trim.wdl]:
```suggestion

```

File: wdl/workflows/pe/trim.wdl [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+            resources=compute.runtimes[machine_size],
+    }
+
+    call adapter_trimmed_fastqs.trim_fastq_pair_to_length {
+        input:
+            fastqs=trim_adapters_on_fastq_pair.trimmed_fastqs,
+            trim_length=trim_length,
+            resources=compute.runtimes[machine_size],
+    }
+
```
Comment (ottojolanki) [inline wdl/workflows/pe/trim.wdl]:
```suggestion

```

File: wdl/subworkflows/filter_bam_reads_with_qc_fail_flag.wdl [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+workflow filter_bam_reads_with_qc_fail_flag {
+    input {
+        File flagged_bam
+        Int? qc_fail_flag = 512
+        Resources resources
+        SamtoolsViewParams params = object {
+            exclude: qc_fail_flag,
+            output_bam: true
+        }
+        String? out_path = "qc_fail_flag_filtered.bam"
```
Comment (ottojolanki) [inline wdl/subworkflows/filter_bam_reads_with_qc_fail_flag.wdl]: Should we call this something like "remove_reads_with_qc_fail_flag" just to make extra sure there is no way to confuse removing reads with flagging reads? I don't feel super strongly about this, just wondering.

Comment (keenangraham) [reply wdl/subworkflows/filter_bam_reads_with_qc_fail_flag.wdl]: We should take back the word filter. Marked versus flagged is probably more confusing.

Comment (ottojolanki) [reply wdl/subworkflows/filter_bam_reads_with_qc_fail_flag.wdl]: so you suggest "marked/flagged (choose one)" == fiddling bam binary flags;
"remove_reads_with_xyz_flag" == actually removing reads? I would like something like this.

**Pull request merged**